src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,838 @@
+    },
+    /**
+     * Determines whether the `v-calendar` will allow a single date/time,
+     * or a range of dates/times.
+     */
+    range: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    /**
+     * A custom set of time frames to be displayed as selectable buttons.
+     * The `timeframeLength`, `start`, and `end` values are passed in as functions,
+     * allowing for on-the-fly date boundary creation.
+     */
+    timePeriods: {
+      type: Array as PropType<Array<TimeFrameSection>>,
+      required: false,
+      default: () => [],
+      validator: (sectionsArray: TimeFrameSection[]) => {
+        return sectionsArray.every((item: TimeFrameSection) => {
+          return Array.isArray(item.values) && item.values.every((timeframe: TimePeriod) => {
+            // Check validity of each timeframe
+            return typeof timeframe.timeframeText === 'string' &&
+              timeframe.timeframeLength !== undefined &&
+              typeof timeframe.key === 'string' &&
+              timeframe.key !== undefined &&
+              typeof timeframe.display === 'string' &&
+              timeframe.display !== undefined &&
+              timeframe.start !== undefined &&
+              timeframe.end !== undefined
+          })
+        })
+      },
+    },
+    /**
+     * Sets the input field to a fixed width
+     */
+    width: {
+      type: String,
+      required: false,
+      default: 'auto',
+    },
+  },
+  emits: ['input', 'change', 'update:modelValue'],
+  setup(props, { emit }) {
+    const modelConfig = { type: 'number' } // https://vcalendar.io/datepicker.html#model-config
+    const calendarSelectAttributes = {
+      highlight: {
+        start: { class: 'vcal-day-start' },
+        base: { class: 'vcal-day-base' },
+        end: { class: 'vcal-day-end' },
+      },
+    }
+    const calendarDragAttributes = {
+      highlight: {
+        start: { class: 'vcal-day-drag-start' },
+        base: { class: 'vcal-day-drag-base' },
+        end: { class: 'vcal-day-drag-end' },
+      },
+    }
+
+    const hasCalendar = computed(() => props.mode !== 'relative')
+    const hasTimePeriods = computed(() => props.timePeriods && props.timePeriods.length)
+    const showCalendar = computed(() => state.tabName === 'custom' || !hasTimePeriods.value)
+    const submitDisabled = computed(() => {
+      // If either the calendar is in range selection mode, or relative time frames
+      // are present, check whether both `start` and `end` are set;
+      // Otherwise, it's a single date or time, so only check `start`
+      return props.range || hasTimePeriods.value
+        ? !state.selectedRange.start || !state.selectedRange.end
+        : !state.selectedRange.start
+    })
+
+    const widthStyle = computed(() => {
+      return {
+        width: props.width === 'auto' || props.width.endsWith('%') || props.width.endsWith('px') ? props.width : props.width + 'px',
+      }
+    })
```

Comment:
Since we're on the beta branch now, can we use the utils helper that computed the width value?